### PR TITLE
Ucred returns pid field for OpenBSD.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

**Comment by Darksonn** [comment]
Is this because it currently does the wrong thing? Can we have a test that catches it?

**Comment by devnexen** [comment]
Not so much but more to address #3102

**Comment by Darksonn** [comment]
Aha, that would be nice to include in the description.

**Comment by Darksonn** [comment]
Please push new commits instead of amending your existing commit. It makes it easier for me to see what changed since I last looked.

**Comment by devnexen** [comment]
was mostly stylistic as I do not have rustfmt on my netbsd VM.

File: tokio/src/net/unix/ucred.rs [hunk 1]
```diff
@@ -48,14 +46,20 @@ pub(crate) use self::impl_macos::get_peer_cred;
...
     use libc::{c_void, getsockopt, socklen_t, SOL_SOCKET, SO_PEERCRED};
     use std::{io, mem};
 
+    #[cfg(target_os = "openbsd")]
+    use libc::sockpeercred;
+    #[cfg(any(target_os = "linux", target_os = "android"))]
     use libc::ucred;
+    #[cfg(target_os = "openbsd")]
+    #[allow(non_camel_case_types)]
+    type ucred = sockpeercred;
```
Comment (Darksonn) [inline tokio/src/net/unix/ucred.rs:62]:
```suggestion
    #[cfg(target_os = "openbsd")]
    use libc::sockpeercred as ucred;
    #[cfg(any(target_os = "linux", target_os = "android"))]
    use libc::ucred;
```

File: tokio/src/net/unix/ucred.rs [hunk 2]
```diff
@@ -97,12 +101,48 @@ pub(crate) mod impl_linux {
...
+                    gid: unpcbid.unp_egid,
+                    pid: Some(unpcbid.unp_pid),
+                })
+            } else {
+                Err(io::Error::last_os_error())
+            }
+        }
+    }
+}
+#[cfg(any(target_os = "dragonfly", target_os = "freebsd"))]
```
Comment (Darksonn) [inline tokio/src/net/unix/ucred.rs:145]: Just inserting a newline here.
```suggestion
}

#[cfg(any(target_os = "dragonfly", target_os = "freebsd"))]
```

**Review by Darksonn — APPROVED** [review]
This seems right. It's a bit unfortunate that we don't have tests for any of this though.

**Pull request merged**